test/cloudtest/pkg/commands/main.go

```diff
@@ -587,48 +619,60 @@ func fromClusterState(state clusterState) string {
 	case clusterShutdown:
 		return "shutdown"
 	}
-	return fmt.Sprintf("unknown state: %v", state)
+	return fmt.Sprintf("unknown state: %v", inst.state)
 }
 
 func (ctx *executionContext) createTasks() {
 	taskIndex := 0
 	for i, test := range ctx.tests {
-		selector := test.ExecutionConfig.ClusterSelector
-		// In case of one cluster, we create task copies and execute on every cloud.
-
-		var task *testTask
-		if test.ExecutionConfig.ClusterCount > 1 {
-			for _, clusterName := range selector {
-				for _, cluster := range ctx.clusters {
-					if clusterName == cluster.config.Name {
-						if task == nil {
-							task = ctx.createSingleTask(taskIndex, test, cluster, i)
-							taskIndex++
-						} else {
-							task.clusters = append(task.clusters, cluster)
-							cluster.tasks[task.test.Key] = task
-						}
-						break
-					}
-				}
+		if test.ExecutionConfig.ConcurrencyRetry > 0 {
+			for j := 0; j < int(test.ExecutionConfig.ConcurrencyRetry); j++ {
+				taskIndex = ctx.createTask(test, taskIndex, i)
+				test.Key = fmt.Sprintf("%s-%d", test.Key, j)
 			}
 		} else {
+			taskIndex = ctx.createTask(test, taskIndex, i)
+		}
+	}
+}
+
+func (ctx *executionContext) createTask(test *model.TestEntry, taskIndex, i int) int {
```

Comment:
Since the code is being refactored, can we give 'i' a descriptive name, like totalTasks?

Reply:
Fixed. Thanks.